hypex/ab_test/limit_distribution.py

```diff
@@ -0,0 +1,203 @@
+    variances = [np.var(sample) * sample_size / (sample_size - 1) for sample in samples] if equal_variance else None
+
+    if quantiles is None:
+        quantiles = quantile_of_marginal_distribution(num_samples=num_samples,
+                                                      quantile_level=1 - significance_level / num_samples,
+                                                      variances=variances)  # квантиль предельного распределения
+    for j in range(num_samples):
+        min_t_value = np.inf
+        for i in range(num_samples):
+            if i != j:
+                t_value = np.sqrt(sample_size) * (means[j] - means[i]) / np.sqrt(variances[j] + variances[i])
+                min_t_value = min(min_t_value, t_value)
+
+        if min_t_value > quantiles[j]:
+            return j + 1
+    return 0
+
+
+def min_sample_size(number_of_samples, minimum_detectable_effect, variances, significance_level=0.05, power_level=0.2,
+                    equal_variance=True, quantile_1=None, quantile_2=None, initial_estimate=None):
+    random_state = 42
+    if equal_variance:
+        if quantile_1 is None:
+            quantile_1 = quantile_of_marginal_distribution(num_samples=number_of_samples,
+                                                           quantile_level=1 - significance_level / number_of_samples)  # квантиль предельного распределения 1-alpha/k
+
+        if quantile_2 is None:
+            quantile_2 = quantile_of_marginal_distribution(num_samples=number_of_samples,
+                                                           quantile_level=power_level)  # квантиль предельного распределения beta
+
+        return int(2 * variances * ((quantile_1 - quantile_2) / minimum_detectable_effect) ** 2) + 1
+    else:
+        iteration_size = 3000  # Количество итераций
+        if quantile_1 is None:
+            quantile_1 = quantile_of_marginal_distribution(num_samples=number_of_samples,
+                                                           quantile_level=1 - significance_level / number_of_samples,
+                                                           variances=variances,
+                                                           )  # набор квантилей предельного распределения
+        sample_sizes = []  # для размеров выборки
+        for sample_index in range(number_of_samples):
+            sample_size = initial_estimate or 0
+            current_power = 0  # мощность
+            while current_power < 1 - power_level:
+                sample_size += 100
+                current_power = 0
+                total_samples = norm.rvs(size=[iteration_size, number_of_samples], random_state=random_state)
+                for sample in total_samples:
+                    min_t_value = np.inf
+                    for i in range(number_of_samples):
+                        if i != sample_index:
+                            t_value = sample[sample_index] / np.sqrt(1 + variances[i] / variances[sample_index]) - \
```

Comment:
Dmatryus told me, that \ for line breaks is not a good pattern. Use () for your formula